File: backend/internal/domain/work_item.go
package domain

import (
	"errors"
	"strings"
	"time"
)

type WorkItemType string
type WorkItemStatus string
type WorkItemPriority string

const (
	WorkItemTypeTask      WorkItemType = "task"
	WorkItemTypeUserStory WorkItemType = "user_story"
	WorkItemTypeBug       WorkItemType = "bug"

	WorkItemStatusToDo       WorkItemStatus = "to_do"
	WorkItemStatusInProgress WorkItemStatus = "in_progress"
	WorkItemStatusReview     WorkItemStatus = "review"
	WorkItemStatusDone       WorkItemStatus = "done"
	WorkItemStatusBlocked    WorkItemStatus = "blocked"

	WorkItemPriorityLow      WorkItemPriority = "low"
	WorkItemPriorityMedium   WorkItemPriority = "medium"
	WorkItemPriorityHigh     WorkItemPriority = "high"
	WorkItemPriorityCritical WorkItemPriority = "critical"
)

type WorkItem struct {
	ID          string
	ProjectID   string
	Title       string
	Description string
	Type        WorkItemType
	Status      WorkItemStatus
	Priority    WorkItemPriority
	AssigneeID  string
	ReporterIDs []string
	StoryPoints int32
	Features    map[string]any
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

func (item WorkItem) Validate() error {
	if strings.TrimSpace(item.ProjectID) == "" {
		return errors.New("project_id is required")
	}
	if strings.TrimSpace(item.Title) == "" {
		return errors.New("title is required")
	}
	if item.Type == "" {
		return errors.New("type is required")
	}
	if item.Status == "" {
		return errors.New("status is required")
	}
	if item.Priority == "" {
		return errors.New("priority is required")
	}
	if item.StoryPoints < 0 {
		return errors.New("story_points cannot be negative")
	}
	return nil
}

File: frontend/src/types/work-item.ts
export type WorkItemType = "story" | "task" | "bug";

export type WorkItemStatus = "todo" | "in_progress" | "review" | "done";

export type Priority = "highest" | "high" | "medium" | "low";

export interface Member {
  id: string;
  name: string;
  initials: string;
  color: string;
}

export interface AcceptanceCriterion {
  text: string;
  done: boolean;
}

export interface WorkItem {
  /** Ticket key, e.g. "CHURA-109". The real API may split this into a UUID `id` plus a human-readable `key`. */
  id: string;
  projectId?: string;
  type: WorkItemType;
  title: string;
  description?: string;
  status: WorkItemStatus;
  blocked: boolean;
  blockedReason?: string;
  priority: Priority;
  storyPoints: number;
  epic: string;
  assigneeId?: string;
  reporterId?: string;
  /** null = unplanned, still sitting in the Backlog */
  sprintId: string | null;
  labels: string[];
  features?: Record<string, unknown>;
  acceptanceCriteria: AcceptanceCriterion[];
  createdAt: string;
  updatedAt: string;
}

export interface Sprint {
  id: string;
  name: string;
  startDate: string;
  endDate: string;
  status: "active" | "planned" | "completed";
}

export interface WorkItemComment {
  id: string;
  workItemId: string;
  authorId: string;
  body: string;
  createdAt: string;
}
